In [78]:
import csv
f = open('guns.csv','r')
file = csv.reader(f)
guns_data = list(file)
print(guns_data[:5])

[['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education'], ['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4']]


In [79]:
header = guns_data[0]
guns_data = guns_data[1:]
print(header)
print(guns_data[:10])

['', 'year', 'month', 'intent', 'police', 'sex', 'age', 'race', 'hispanic', 'place', 'education']
[['1', '2012', '01', 'Suicide', '0', 'M', '34', 'Asian/Pacific Islander', '100', 'Home', '4'], ['2', '2012', '01', 'Suicide', '0', 'F', '21', 'White', '100', 'Street', '3'], ['3', '2012', '01', 'Suicide', '0', 'M', '60', 'White', '100', 'Other specified', '4'], ['4', '2012', '02', 'Suicide', '0', 'M', '64', 'White', '100', 'Home', '4'], ['5', '2012', '02', 'Suicide', '0', 'M', '31', 'White', '100', 'Other specified', '2'], ['6', '2012', '02', 'Suicide', '0', 'M', '17', 'Native American/Native Alaskan', '100', 'Home', '1'], ['7', '2012', '02', 'Undetermined', '0', 'M', '48', 'White', '100', 'Home', '2'], ['8', '2012', '03', 'Suicide', '0', 'M', '41', 'Native American/Native Alaskan', '100', 'Home', '2'], ['9', '2012', '02', 'Accidental', '0', 'M', '50', 'White', '100', 'Other specified', '3'], ['10', '2012', '02', 'Suicide', '0', 'M', 'NA', 'Black', '998', 'Home', '5']]


In [80]:
years = [row[1] for row in guns_data]
year_counts = {}
for year in years:
    if year in year_counts:
        year_counts[year] += 1
    else:
        year_counts[year] = 1
print(year_counts)

{'2014': 33599, '2012': 33563, '2013': 33636}


In [81]:
import datetime
dates = [datetime.datetime(int(row[1]),int(row[2]),day=1) for row in guns_data]
date_counts = {}
for date in dates:
    if date in date_counts:
        date_counts[date] += 1
    else:
        date_counts[date] = 1
print(date_counts)

{datetime.datetime(2013, 8, 1, 0, 0): 2859, datetime.datetime(2013, 7, 1, 0, 0): 3079, datetime.datetime(2012, 3, 1, 0, 0): 2743, datetime.datetime(2014, 3, 1, 0, 0): 2684, datetime.datetime(2012, 7, 1, 0, 0): 3026, datetime.datetime(2014, 9, 1, 0, 0): 2914, datetime.datetime(2014, 7, 1, 0, 0): 2884, datetime.datetime(2013, 4, 1, 0, 0): 2798, datetime.datetime(2013, 2, 1, 0, 0): 2375, datetime.datetime(2013, 12, 1, 0, 0): 2765, datetime.datetime(2014, 1, 1, 0, 0): 2651, datetime.datetime(2012, 6, 1, 0, 0): 2826, datetime.datetime(2014, 11, 1, 0, 0): 2756, datetime.datetime(2013, 3, 1, 0, 0): 2862, datetime.datetime(2012, 9, 1, 0, 0): 2852, datetime.datetime(2013, 10, 1, 0, 0): 2808, datetime.datetime(2014, 10, 1, 0, 0): 2865, datetime.datetime(2014, 2, 1, 0, 0): 2361, datetime.datetime(2012, 1, 1, 0, 0): 2758, datetime.datetime(2014, 4, 1, 0, 0): 2862, datetime.datetime(2013, 5, 1, 0, 0): 2806, datetime.datetime(2012, 10, 1, 0, 0): 2733, datetime.datetime(2012, 4, 1, 0, 0): 2795, datet

In [82]:
sex_counts = {}
for row in guns_data:
    sex = row[5]
    if sex in sex_counts:
        sex_counts[sex] += 1
    else:
        sex_counts[sex] = 1
        
race_counts = {}
for row in guns_data:
    race = row[7]
    if race in race_counts:
        race_counts[race] += 1
    else:
        race_counts[race] = 1
print(race_counts, sex_counts)
        

{'Asian/Pacific Islander': 1326, 'Black': 23296, 'White': 66237, 'Hispanic': 9022, 'Native American/Native Alaskan': 917} {'F': 14449, 'M': 86349}


In [83]:
f = open('census.csv', 'r')
file = csv.reader(f)
census = list(file)
print(census)


[['Id', 'Year', 'Id', 'Sex', 'Id', 'Hispanic Origin', 'Id', 'Id2', 'Geography', 'Total', 'Race Alone - White', 'Race Alone - Hispanic', 'Race Alone - Black or African American', 'Race Alone - American Indian and Alaska Native', 'Race Alone - Asian', 'Race Alone - Native Hawaiian and Other Pacific Islander', 'Two or More Races'], ['cen42010', 'April 1, 2010 Census', 'totsex', 'Both Sexes', 'tothisp', 'Total', '0100000US', '', 'United States', '308745538', '197318956', '44618105', '40250635', '3739506', '15159516', '674625', '6984195']]


In [84]:
mapping = {}
census = census[1]
mapping['Asian/Pacific Islander'] = int(census[14]) + int(census[15])

mapping['Black'] = int(census[12])

mapping['Hispanic'] = int(census[11])

mapping['Native American/Native Alaskan'] = int(census[13])

mapping['White'] = int(census[10])

print(mapping)
print(race_counts)


{'Asian/Pacific Islander': 15834141, 'Black': 40250635, 'White': 197318956, 'Hispanic': 44618105, 'Native American/Native Alaskan': 3739506}
{'Asian/Pacific Islander': 1326, 'Black': 23296, 'White': 66237, 'Hispanic': 9022, 'Native American/Native Alaskan': 917}


In [85]:
race_per_hundredk = {}
for race in race_counts:
    total = (race_counts[race] / mapping[race]) * 100000
    race_per_hundredk[race] = total
print(race_per_hundredk)

{'Asian/Pacific Islander': 8.374309664161762, 'Black': 57.8773477735196, 'White': 33.56849303419181, 'Hispanic': 20.220491210910907, 'Native American/Native Alaskan': 24.521955573811088}


In [87]:
intents = [gun_death[3] for gun_death in guns_data]
races = [gun_death[7] for gun_death in guns_data]
homicide_race_counts = {}
    

In [88]:
for i,race in enumerate(races):
    if intents[i] == 'Homicide':
        if race in homicide_race_counts:
            homicide_race_counts[race] += 1
        else:
            homicide_race_counts[race] = 1
print(homicide_race_counts)

{'Asian/Pacific Islander': 559, 'Black': 19510, 'White': 9147, 'Hispanic': 5634, 'Native American/Native Alaskan': 326}


In [90]:
for race in homicide_race_counts:
    total = (homicide_race_counts[race] / mapping[race]) * 100000
    homicide_race_counts[race] = total

print(homicide_race_counts)    

{'Asian/Pacific Islander': 3.530346230970155, 'Black': 48.471284987180944, 'White': 4.6356417981453335, 'Hispanic': 12.627161104219914, 'Native American/Native Alaskan': 8.717729026240365}


# __Conclusion__

### Findings

Based on my analysis on gun deaths between 2012 and 2014, it appears that _Black/African-American_ individuals are killed by guns the most in America per 100,000 (57.8 people), followed by _White_ individuals (33.6 people). Similarly, when looking at homicide-only gun deaths _Black/African-American_ individuals were killed four times more per 100,000 (48.5 people) than the next highest demographic (_Hispanic_: 12.6 people).

### Possible Next Steps

After this initial analysis of gun deaths in America from 2012 to 2014, I would like to explore the education level and age of the victims to see what percentage of them had were in grade school at the time of death. Dividing this number into racial categories can also paint a picture into which demographic was exposed to guns more during adolescence from 2012-2014.

Exploring age, race, and education level in suicide-victims, both exclusively and together, is another step I would like to persue as that analysis could generate a high-level insight into the likelihood of suicide 